### PR TITLE
MERC-2617 Region manifest

### PR DESCRIPTION
**Pull request by mcampa**
#### What?

Stencil CLI needs to produce a list of regions within each of the template files.

Here I use regex to scan for region tags, building a full template per page to "guaranty" the order of the list returned.

I wrote a test case that considers "all" cases using `dynamicComponent` and `partial/block` helpers

@pedelman @mattolson @mjschock

### CODE REVIEW

**Comment by mcampa** [comment]
Forgot to ping @tabayomi

File: lib/stencil-bundle.js [hunk 1]
```diff
@@ -198,30 +221,31 @@ Bundle.prototype.getJspmBundleTask = function (jspmConfig) {
...
 
-        filePaths = filePaths.map((file) => {
-            return upath.toUnix(file.replace(this.templatesBasePath + Path.sep, '').replace('.html', ''));
+        const partials = filePaths.map(file => {
+            return Upath.toUnix(file.replace(this.templatesPath + Path.sep, '').replace('.html', ''));
         });
 
-        output.templates = filePaths;
-        this.manifest = output;
+        taskResults.manifest = {
```
Comment (mjschock) [inline lib/stencil-bundle.js]: to keep it a bit cleaner, could we callback with taskResults instead?

File: lib/regions.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        return buildTemplateIncludes(templates, partial, visited);
+    });
+}
+
+/**
+ * Parse a template content and return an array of unique region objects
+ * @param {String} content
+ * @return {Array}
+ */
+function parseRegions(content) {
```
Comment (pedelman) [inline lib/regions.js]: Can we add some test coverage for the regex. Its hard to review whether this regex handles all the special cases.

Comment (mcampa) [inline lib/regions.js]: Can you provide the special cases?

File: lib/regions.spec.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            '{{{  region name="foobar1"  }}}': [{ name: 'foobar1' }],
+            '{{{  region  name=" "  }}}': [],
+            '{{{  region  name="invalid name"  }}}': [],
+            '{{  region  name="two_brackets"  }}': [],
+            '{{{  region   name=\'foobar\'  }}}': [{ name: 'foobar' }],
+            '{{{region name="foobar" type="widget"}}}': [{ name: 'foobar' }],
+            '{{{ region type="widget"  name="foobar"  }}}': [{ name: 'foobar' }],
+            '{{{ region name="foobar"  type="widget" }}}': [{ name: 'foobar' }],
+            '{{{ region name=\'foo\' }}} \n {{{ region name="bar" }}}': [{ name: 'foo' }, { name: 'bar' }],
+            '{{{region name=\'foo\'}}}{{{region name="bar"}}}{{{region name="foo"}}}': [{ name: 'foo' }, { name: 'bar' }],
```
Comment (mcampa) [inline lib/regions.spec.js]: @pedelman let me know if this is enough

**Pull request merged**